### PR TITLE
Update Kotlin to 1.4.30

### PR DESCRIPTION
**Pull request by antoinelamy**
## Description
- Migrate kotlin to `1.4.30`
- Remove deprecated `jcenter` repository
- Migrate Gradle scripts to use Kotlin DSL over Groovy
- Update dependencies
- Bump major version as it contains a breaking change

## Motivation and Context
Make Trikot available for Kotlin `1.4.30`

## How Has This Been Tested?
N/A

## Types of changes
Breaking change

### CODE REVIEW

**Comment by antoinelamy** [comment]
The CI failure is expected because version 2.0.0 needs to be releases in order to be able to fix the ktx modules.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,6 +17,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+        with:
```
Comment (marclefrancois) [inline .github/workflows/ci.yml:20]: ça doit accélérer pas pire ça ?

**Review by marclefrancois — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (antoinelamy) [reply .github/workflows/ci.yml:20]: > It must speed things up, does it?
The repos are still fairly small with no image resources so it doesn't make a lot of difference but it's a good thing for the future.

**Review by martingagnon — COMMENTED** [review]
Late 👍

**Pull request merged**